crates/api/src/space.rs

```diff
@@ -57,6 +58,9 @@ pub trait Space: 'static + Send + Sync + std::fmt::Debug {
     /// be made to publish this tombstone to peers in the space as well.
     fn local_agent_leave(&self, local_agent: id::AgentId) -> BoxFut<'_, ()>;
 
+    /// Get a list of all local agents currently joined to this space.
+    fn get_local_agents(&self) -> BoxFut<'_, K2Result<Vec<DynLocalAgent>>>;
```

Comment:
Re my previous comment, it makes sense gossip would need this... but we may need to make a separate data structure that the space can own but that can also be passed into gossip without causing circular references.